### PR TITLE
Adding some integration tests

### PR DESCRIPTION
**Pull request by corylanou**

### CODE REVIEW

File: cmd/influxd/run/server.go [hunk 1]
```diff
@@ -144,7 +143,6 @@ func (s *Server) Close() error {
 type Service interface {
 	Open() error
 	Close() error
-	Addr() net.Addr
```
Comment (otoolep) [inline cmd/influxd/run/server.go]: Why this change? I don't object to it, but we just discussed it this morning.

I do think that `Addr` may not be the right call to have on `Service` so it's OK with me.

Comment (corylanou) [inline cmd/influxd/run/server.go]: Turns out we don't need it.  We only need it for integration testing, and we can do type asserts in the test if we actually need it.  This makes `Services` all act/feel the same from the Server standpoint.

Comment (otoolep) [inline cmd/influxd/run/server.go]: So *this* is the new definition of the `Service` interface? Makes sense to me.

Comment (corylanou) [inline cmd/influxd/run/server.go]: Yes, this is the new definition now.
